src/value.c

```diff
@@ -192,7 +229,64 @@ size_t SIValue_StringConcat(SIValue *strings, unsigned int string_count, char *b
 	return offset;
 }
 
+SIValue SIValue_ConcatString(const SIValue a, const SIValue b) {
+	SIValue result;
+	char buffer[512];
+	char *string_arg = NULL;
+	// a is not a string - scalar + string
+	if(a.type != T_STRING) {
+		/* a is numeric, convert to string. */
+		SIValue_ToString(a, buffer, 512);
+		result = SI_DuplicateStringVal(buffer);
+	}
+	// string + value
+	else result = SI_Clone(a);
+
+	unsigned int argument_len = 0;
+	if(b.type != T_STRING) {
+		/* b is not a numeric, get a string representation. */
+		argument_len = SIValue_ToString(b, buffer, 512);
+		string_arg = buffer;
+	} else {
+		string_arg = b.stringval;
+		argument_len = strlen(string_arg);
+	}
+
+	/* Concat, make sure result has enough space to hold new string. */
+	unsigned int required_size = strlen(result.stringval) + argument_len + 1;
+	result.stringval = rm_realloc(result.stringval, required_size);
+	strcat(result.stringval, string_arg);
+	return result;
+}
+
+SIValue SIValue_ConcatList(const SIValue a, const SIValue b) {
```

Comment:
// Assumption either 'a' or 'b' is a list.

Reply:
done